Add torch-pme recipe

PicoCentauri opened the pull request:
<!--
Thank you very much for putting in this recipe PR!

This repository is very active, so if you need help with a PR, please let the
right people know. There are language-specific teams for reviewing recipes.

| Language        | Name of review team           |
| --------------- | ----------------------------- |
| python          | `@conda-forge/help-python`    |
| python/c hybrid | `@conda-forge/help-python-c`  |
| r               | `@conda-forge/help-r`         |
| java            | `@conda-forge/help-java`      |
| nodejs          | `@conda-forge/help-nodejs`    |
| c/c++           | `@conda-forge/help-c-cpp`     |
| perl            | `@conda-forge/help-perl`      |
| Julia           | `@conda-forge/help-julia`     |
| ruby            | `@conda-forge/help-ruby`      |
| Rust            | `@conda-forge/help-rust`      |
| Go              | `@conda-forge/help-go`        |
| other           | `@conda-forge/staged-recipes` |

Once the PR is ready for review, please mention one of the teams above in a
new comment. i.e. `@conda-forge/help-some-language, ready for review!`
Then, a bot will label the PR as 'review-requested'.

Due to GitHub limitations, first time contributors to conda-forge are unable
to ping conda-forge teams directly, but you can [ask a bot to ping the team][1]
using a special command in a comment on the PR to get the attention of the
`staged-recipes` team. You can also consider asking on our [Zulip chat][2]
if your recipe isn't reviewed promptly.

[1]: https://conda-forge.org/docs/maintainer/infrastructure.html#conda-forge-admin-please-ping-team
[2]: https://conda-forge.zulipchat.com

All apologies in advance if your recipe PR does not receive prompt attention.
This is a high volume repository and the reviewers are volunteers. Review times
vary depending on the number of reviewers on a given language team and may be days
or weeks. We are always looking for more staged-recipe reviewers. If you are
interested in volunteering, please contact a member of @conda-forge/core.
We'd love to have the help!
-->

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I failed to even lint the recipe, probably because of a conda-smithy bug :cry:. This likely indicates a problem in your `meta.yaml`, though. To get a traceback to help figure out what's going on, install conda-smithy and run `conda smithy recipe-lint --conda-forge .` from the recipe directory. You can also examine the [workflow logs](https://github.com/conda-forge/conda-forge-webservices/actions/runs/12233294018) for more detail.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/12233294018](https://github.com/conda-forge/conda-forge-webservices/actions/runs/12233294018). Examine the logs at this URL for more detail.</sub>

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/torch-pme/meta.yaml```) and found it was in an excellent condition.

I do have some suggestions for making it better though...


For **recipes/torch-pme/meta.yaml**:

 * ℹ️ `noarch: python` recipes should usually follow the syntax in our [documentation](https://conda-forge.org/docs/maintainer/knowledge_base/#noarch-python) for specifying the Python version.
   - For the `test.requires` section of the recipe, you should usually use `python {{ python_min }}` for the `python` entry.
   - If the package requires a newer Python version than the currently supported minimum version on `conda-forge`, you can override the `python_min` variable by adding a Jinja2 `set` statement at the top of your recipe (or using an equivalent `context` variable for v1 recipes).
 * ℹ️ PyPI default URL is now pypi.org, and not pypi.io. You may want to update the default source url.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/12233338125](https://github.com/conda-forge/conda-forge-webservices/actions/runs/12233338125). Examine the logs at this URL for more detail.</sub>

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/torch-pme/meta.yaml```) and found some lint.

Here's what I've got...


For **recipes/torch-pme/meta.yaml**:

 * ❌ Invalid `noarch` value `python 9999`. Should be one of `python`, `generic`.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/12233420711](https://github.com/conda-forge/conda-forge-webservices/actions/runs/12233420711). Examine the logs at this URL for more detail.</sub>

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/torch-pme/meta.yaml```) and found it was in an excellent condition.

Luthaf commented:
I'm happy to help maintain this package!

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/torch-pme/meta.yaml```) and found some lint.

Here's what I've got...


For **recipes/torch-pme/meta.yaml**:

 * ❌ `noarch` packages can't have skips with selectors. If the selectors are necessary, please remove `noarch: python`.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/12237576816](https://github.com/conda-forge/conda-forge-webservices/actions/runs/12237576816). Examine the logs at this URL for more detail.</sub>

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/torch-pme/meta.yaml```) and found it was in an excellent condition.

PicoCentauri commented:
@conda-forge/help-python, ready for review!

ocefpaf commented on recipes/torch-pme/meta.yaml:
```diff
@@ -0,0 +1,54 @@
+{% set version = "0.1.0" %}
+
+package:
+  # use name with a hyphen to be consistent with PyPI name
```
```suggestion
```

ocefpaf commented on recipes/torch-pme/meta.yaml:
```diff
@@ -0,0 +1,54 @@
+{% set name = "torch_pme" %}
```
This is the name and should be the same as PyPI.

```suggestion
{% set name = "torch-pme" %}
```

ocefpaf commented on recipes/torch-pme/meta.yaml:
```diff
@@ -0,0 +1,54 @@
+  version: {{ version }}
+
+source:
+  url: https://pypi.org/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
```
```suggestion
  url: https://pypi.org/packages/source/{{ name[0] }}/{{ name }}/torch_pme-{{ version }}.tar.gz
```

ocefpaf commented on recipes/torch-pme/meta.yaml:
```diff
@@ -0,0 +1,54 @@
+  sha256: 7d31801b44a19a1633ecc4f2c484811342940b074dcd01f30dd97ff5a4e69278
+
+build:
+  # conda-forge's pytorch is not yet available on Windows
+  skip: true  # [win or py<39]
```
It is OK for noarch package to fails on Windows with a missing dependency. We prefer noarch b/c it is faster to build and the package will be ready when the missing dependency is available on Windows. 

```suggestion
  noarch: python
```

ocefpaf commented on recipes/torch-pme/meta.yaml:
```diff
@@ -0,0 +1,54 @@
+requirements:
+  host:
+    - pip
+    - python
```
```suggestion
    - python {{ python_min }}
```

ocefpaf commented on recipes/torch-pme/meta.yaml:
```diff
@@ -0,0 +1,54 @@
+    - setuptools
+    - wheel
+  run:
+    - python
```
```suggestion
    - python >={{ python_min }}
```

ocefpaf commented on recipes/torch-pme/meta.yaml:
```diff
@@ -0,0 +1,54 @@
+    - torchpme
+  commands:
+    - pip check
+  requires:
+    - pip
```
```suggestion
  requires:
    - python {{ python_min }}
    - pip
```

ocefpaf replied:
Ping

ocefpaf reviewed (commented):
Please use grayskull to generate Python recipes. Grayskull make it easier for us to review.

See https://github.com/conda-forge/staged-recipes?tab=readme-ov-file#grayskull---recipe-generator-for-python-packages-on-pypi for more information.

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/torch-pme/meta.yaml```) and found some lint.

Here's what I've got...


For **recipes/torch-pme/meta.yaml**:

 * ❌ Invalid `noarch` value `python 9999`. Should be one of `python`, `generic`.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/12254778571](https://github.com/conda-forge/conda-forge-webservices/actions/runs/12254778571). Examine the logs at this URL for more detail.</sub>

PicoCentauri commented:
@ocefpaf thanks for the comments and the suggestions of Grayskull. I was not aware that this tool exists. This will make creation much easier in the future. I hope I addressed all suggestions.

ocefpaf commented on recipes/torch-pme/meta.yaml:
```diff
@@ -0,0 +1,52 @@
+build:
+  noarch: python
+  number: 0
+  script: {{ PYTHON }} -m pip install . -vv
```
```suggestion
  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
```

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/torch-pme/meta.yaml```) and found it was in an excellent condition.

I do have some suggestions for making it better though...


For **recipes/torch-pme/meta.yaml**:

 * ℹ️ `noarch: python` recipes should usually follow the syntax in our [documentation](https://conda-forge.org/docs/maintainer/knowledge_base/#noarch-python) for specifying the Python version.
   - For the `run` section of the recipe, you should usually use `python >={{ python_min }}` for the `python` entry.
   - For the `test.requires` section of the recipe, you should usually use `python {{ python_min }}` for the `python` entry.
   - If the package requires a newer Python version than the currently supported minimum version on `conda-forge`, you can override the `python_min` variable by adding a Jinja2 `set` statement at the top of your recipe (or using an equivalent `context` variable for v1 recipes).


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/12254828762](https://github.com/conda-forge/conda-forge-webservices/actions/runs/12254828762). Examine the logs at this URL for more detail.</sub>

ocefpaf commented on recipes/torch-pme/meta.yaml:
```diff
@@ -0,0 +1,53 @@
+{% set name = "torch-pme" %}
+{% set version = "0.1.0" %}
+{% set python_min = "3.9" %}
```
```suggestion
```

ocefpaf commented:
Please don't request another review until you acted on the past one. You still need to add the python min to the test area as per my code suggestion from [before](https://github.com/conda-forge/staged-recipes/pull/28488#discussion_r1876284621)

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/torch-pme/meta.yaml```) and found it was in an excellent condition.

PicoCentauri commented:
Sorry, I thought this was not necessary because it is not explicitly required by the tests. I updated the branch.

ocefpaf commented:
> Sorry, I thought this was not necessary because it is not explicitly required by the tests. I updated the branch.

It was in the linter notes, in your review, and I pinged you 2 times. Please trust the reviewers, that will save us a lot of our volunteered time.

PicoCentauri commented:
Yes sorry again and thanks for the time. Really appreciate the work.
